doc/contribute/coding_guidelines/index.rst

```diff
@@ -944,3 +944,92 @@ Related GitHub Issues and Pull Requests are tagged with the `Inclusive Language
+project. It is used to automate compliance with a range of coding and security
+standards.
+Currently, the tool is set to MISRA C:2012 Coding Standard because Zephyr
+:ref:`coding_guidelines` are based on that standard.
+It is used together with the Coverity Scan tool to achieve the best code health
+and precision in bug findings.
+
+Violations fixing process
+=========================
+
+Step 1
+  Any Zephyr Project member, company or a developer can request a license
+  to access Codescan online dashboard.
+
+Step 2
+  A developer starts to review raised violations.
+
+Step 3
+  A developer submits a Github PR with the fix. Commit messages should follow
+  the same guidelines as other PRs in the Zephyr project. Please add a comment
+  that your fix was found by a static coding scanning tool.
+  A developer should follow and refer to the Zephyr :ref:`coding_guidelines`
+  as the basic coding rules. These rules are based on the MISRA C standard.
+
+  Below you can find an example of a recommended commit message::
+
+     lib: os: add braces to 'if' statements
+
+
+     An 'if' (expression) construct shall be followed by a compound statement.
+     Add braces to improve readability and maintainability.
+
+     Found as a coding guideline violation (Rule 15.6) by static
+     coding scanning tool.
+
+     Signed-off-by: Johnny Developer johnny.developer@company.com
+
+Step 4
+  If a violation is a false-positive developer must submit a PR with
+  a suppression tag and an explanation comment.
+
+  The template structure of the comment and tag in the code should be::
+
+     /* Explain why that part of the code is healthy and raised violation is
+      * a false-positive
+      */
+     /* parasoft suppress item Rule ID */
+
+  Below you can find an example of a recommended commit message::
+
+     testsuite: suppress usage of setjmp in a testcode (rule 21.4)
+
+
+     According to the Rule 21.4 the standard header file <setjmp.h> shall not
+     be used. Suppress it, because it raises violation in a testcode,
+     not in a runtime code. Tag suppresses reporting of violation for the
+     current file, starting from the line where the suppression is located.
+
+     Found as a coding guideline violation (Rule 21.4) by static coding
+     scanning tool.
+
+     Signed-off-by: Johnny Developer johnny.developer@company.com
+
+  The example below demonstrates how false positives can be suppressed::
+
+     /* Static code analysis tool can raise a violation that the standard header
+      * <setjmp.h> shall not be used.
+      *
+      * setjmp is using in a test code, not in a runtime code
+      */
+     /* parasoft suppress item MISRAC2012-RULE_21_4-a */
+     /* parasoft suppress item MISRAC2012-RULE_21_4-b */
+
+  The Parasoft tags will suppress all violations of the specific rule until
+  the end of the file. To read more about Suppression Findings in the
+  Parasoft tool refer to the official Parasoft `documentation`_
```

Comment:
>suppress all violations of the specific rule until
  the end of the file

Is this really what we want?

As per the [documentation](https://docs.parasoft.com/display/CPPTEST1031/Suppressing+Findings), `parasoft suppress` and `parasoft-suppress` are not the same -- the former does what is described here, whereas the latter only suppresses the specified rule for the current (following) line.

FWIW, I think the scope of suppression should be limited to the place where the `parasoft` tag is specified; otherwise, this might have an unintended consequence of suppressing other unsanctioned violations in the same file.

cc @AmberHibberd @ioannisg

Reply:
Also, how does Parasoft Codescan handle multiple suppressions with `parasoft-suppress`?

For instance, if we have:
```
/* parasoft-suppress item MISRAC2012-RULE_21_4-a */
/* parasoft-suppress item MISRAC2012-RULE_21_4-b */
SOME CODE
```

Are both `MISRAC2012-RULE_21_4-a` and `MISRAC2012-RULE_21_4-b` suppressed for `SOME CODE`? or only `MISRAC2012-RULE_21_4-b`?

Reply:
Yes, about suppression range necessary to think together. About comment, I think better to have standalone comment.

Reply:
> I think better to have standalone explanation comment.

Sure.
```
/* parasoft-suppress MISRAC2012-RULE_21_4-a "Comment explaining rationale behind this suppression" */
/* parasoft-suppress MISRAC2012-RULE_21_4-b "Comment explaining rationale behind this suppression" */
SOME CODE
```

Reply:
I don't know if it supports doing like that. @ceolin told me, he has access to the Parasoft directly and can try to share with me an access. I can only access dashboard with the results, but can't run tool to see how it detects such kind of tags.

Reply:
> > I think better to have standalone explanation comment.
> 
> Sure (assuming Parasoft Codescan supports the following).
> 
> ```
> /* parasoft-suppress MISRAC2012-RULE_21_4-a "Comment explaining rationale behind this suppression" */
> /* parasoft-suppress MISRAC2012-RULE_21_4-b "Comment explaining rationale behind this suppression" */
> SOME CODE
> ```

I don't know if it supports doing like that. @ceolin told me, he has access to the Parasoft directly and can try to share with me an access. I can only access dashboard with the results, but can't run tool to see how it detects such kind of tags.

Reply:
@stephanosio you can search on Github "parasoft-suppress", it can give a baseline for use tags, like other developers are doing. I'm reviewing it right now.

Reply:
@stephanosio Updated my PR.
1. You are right, better to use parasoft tag that can suppress only 1 line of the code.
`violation line of the code /* parasoft-suppress MISRAC2012-RULE_21_4-a*/`

I see that other developers keep that tag in one line even if it is too long.

2. According to the idea adding a tag with a comment:
`CODE LINE parasoft-suppress MISRAC2012-RULE_21_4-b "Comment explaining rationale behind this suppression" */`

Explanation comment can be really long, and probably it can't fit into one line. I see that no one is doing multiline comments using parasoft tags. So I suggest better keep comment separate (above the violation code line) and the comment should have the same style as other comments in the Zephyr project.

Reply:
Also I created a forum topic on the Parasoft webpage. But seems their forum is dead...
https://forums.parasoft.com/discussion/5320/multiline-parasoft-tags-to-suppress-violations

Reply:
I agree keeping the rationale separated of the suppression.